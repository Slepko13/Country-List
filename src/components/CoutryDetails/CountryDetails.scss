.CountryDetails {
  @media screen and (max-width: 991px) {
    display: none;
  }
  border-radius: 15px;
  height: 100%;
  display:flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  margin-left: 40px;
  .choose__country {
    text-align: center;
    .choose__title  {
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: 84px;
      line-height: 98px;
    }
  }
  .country {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    background-color: #FFFFFF;
    box-shadow: 0px 4px 30px rgba(0, 0, 0, 0.05);
    border-radius: 15px;
    .country__top {
      position:relative;
      height: 188px;
      width:  100%;
      background: url("../../assets/images/flags.png") center center no-repeat;
      background-size: 100% 100%;
      .top__flag {
        position: absolute;
        width: 127px;
        height: 75px;
        top : 10px;
        left : 50%;
        transform: translateX(-50%);
        box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.2), -1px -1px 4px rgba(0, 0, 0, 0.2);
        //object-fit: contain;
        background: center center no-repeat;
        background-size: cover;
      }
    }
    .country__block {
      flex:2;
      position: relative;
      .block__image {
        @media screen and (max-width: 1439px){
          height: 260px;
          width: 315px;
        }
        height: 345px;
        width: 420px;
        position: absolute;
        left : 50%;
        top: 50%;
        transform: translate(-50%, -50%);
      }
      .block__info {
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        .one,
        .five {
          display: flex;
          justify-content: center;
        }
        .two,
        .four{
          display: flex;
          justify-content: space-around;
          .left {
            margin-left: -50px;
          }
        }
        .three {
          display: flex;
          justify-content: space-between;
        }
      }
    }
  }
}