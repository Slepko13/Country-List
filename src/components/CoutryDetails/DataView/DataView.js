import React from 'react';
import './DataView.scss';
import PropTypes from "prop-types";

const DataView = (props) => {
    let {title, data, position, addictions, maxWidth} = props;
    data = data === "" ? "n/a" : data === "Kiev" ? "Kyiv" : data;
    return (
        <div className={position ?
            `DataView ${position}` :
            "DataView"}
             data-testid="position"
        >
            <div className="info__title">{title}</div>
            <div className="info__content" style={{maxWidth}}>{data}<span style={{marginLeft: "10px"}}>{addictions}</span></div>
        </div>
    )
}
export default DataView;

DataView.propTypes = {
    title: PropTypes.string,
    data: PropTypes.string,
    position: PropTypes.string,
    addictions: PropTypes.string,
    maxWidth: PropTypes.string
}