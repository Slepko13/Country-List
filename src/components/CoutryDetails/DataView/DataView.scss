.DataView {
  margin-left: 20px;
  z-index: 5;
  @media screen and (max-width: 991px) {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin-bottom: 20px;

  }
    .info__title{
      font-family: "Roboto";
      font-style: normal;
      font-weight: 300;
      font-size: 14px;
      line-height: 16px;
      text-align: center;

    }
    .info__content {
      font-family: "Roboto";
      font-style: normal;
      font-weight: bold;
      text-align: center;
      color: #313131;
      line-height: 30px;
      font-size: 26px;
      max-height: 30px;
      text-overflow:ellipsis;
      overflow: hidden;
      max-width: 200px;

      &:hover {
        overflow: visible;
        max-width: 800px;
        background-color: #ffffff;
      }
    }
  }

