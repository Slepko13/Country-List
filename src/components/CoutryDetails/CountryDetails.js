import React from 'react';
import './CountryDetails.scss';
import DataView from "./DataView/DataView";

import {  useQuery} from '@apollo/client';

import flags from '../../assets/images/flags.png';
import earth from '../../assets/images/happy_earth.png';
import earthFull from '../../assets/images/happy_earth_ellipses.png'

import {COUNTRY} from './query';
import DataViewLanguages from "./DataViewLanguages/DataViewLanguages";
import DataViewList from "./DataViewList/DataViewList";



const CountryDetails = ({id}) => {

    const { loading, error, data } = useQuery(COUNTRY,{
        variables: {name: id}
    });
    if (loading) return <div className="CountryDetails "><p className="loading">Loading...</p></div>;
    if (error) return <div className="CountryDetails "><p className="error">Error :(</p></div>;

    if(!id)  return (
        <div className="CountryDetails " >
          <div className="choose__country">
            <div className="choose__image">
                <img src={earth} width={266} height={266} alt="happy earth"/>
            </div>
              <div className="choose__title">Choose a card :)</div>
          </div>
        </div>
    )

const{
    name,
    population,
    capital,
    flag:{svgFile :flag},
    subregion,
    currencies,
    timezones,
    callingCodes,
    officialLanguages
} = data.Country[0];

const region = subregion ? subregion.region.name : "n/a";
// const currency = currencies.length ? currencies : "n/a";
// const callCode = callingCodes.length ? callingCodes : "n/a";
// const officialLanguage = officialLanguages.length ? officialLanguages : "n/a";
let  popul, unit;

if (population > 1000000) {
    popul = (Math.round(population/10000)/100).toLocaleString("pl");
    unit = "m";
} else {
    popul = (Math.round(population/10)/100).toLocaleString("pl");
    unit = "t"
}
    return (
        <div className="CountryDetails ">
            <div className="country">
                <div className="country__top">
                    <div className="top__flag" style={{backgroundImage : `url(${flag})`}}></div>
                </div>
                <div className="country__block"
                     data-testid = "country"
                >
                    <img  className="block__image" src={earthFull} />
                    <div className="block__info">
                        <div className="info one">
                            <DataView
                                title="Country"
                                data={name}
                                position=""
                            />
                        </div>
                        <div className="info two"
                             data-testid="data">
                            <DataView
                                title="Capital"
                                data={capital}
                                position="left"
                            />
                            <DataView
                                title="Region"
                                data={region}
                                position="right"
                            />
                        </div>
                        <div
                            data-testid="unit"
                            className="info three">
                            <DataView
                                title="Population"
                                data={popul}
                                position=""
                                addictions={unit}
                            />
                            <DataViewList
                                        title="Time zone"
                                        data={timezones}
                                        position=""
                                        addictions=""
                                    />
                            </div>
                        <div className="info four">
                            <DataViewList
                                title="Currencies"
                                data={currencies}
                                position="left"
                                addictions=""
                            />
                            <DataViewList
                                title="Calling codes"
                                data={callingCodes}
                                position=""
                                addictions="+"
                            />
                        </div>
                        <div className="info five">
                            <DataViewLanguages
                                title="Official languages"
                                data={officialLanguages}
                                direction="row"
                                position=""
                                addictions=""
                            />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )

}

export default CountryDetails;