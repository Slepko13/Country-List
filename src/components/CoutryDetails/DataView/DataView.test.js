import React from "react";
import DataView from "./DataView";
import { act, render,cleanup, screen } from '@testing-library/react';
import {MockedProvider} from "@apollo/client/testing";
import CountryDetails from "../CountryDetails";

const setUp = (props) => shallow(<DataView {...props}/>)

describe("my DataView component", () => {


    it("should render DataView  Component with props(snapshot)", () => {
        let component = shallow(<DataView title="Capital" data="London" position="" addictions=""/>);
        expect(component).toMatchSnapshot();
    });

    it("should render DataView  Component without props(snapshot)", () => {
        let component = shallow(<DataView />);
        expect(component).toMatchSnapshot();
    });

    it("should render DataView  additional classname depending on props ", () => {
        const { container, getByTestId } = render(
                <DataView position="left" />
        );
        expect(getByTestId('position')).toHaveClass('left');
    });

    it("should render DataView without additional classname depending on props ", () => {
        const { container, getByTestId } = render(
            <DataView position={null} />
        );
        expect(container.firstChild.classList.contains('left')).toBe(false)
    });


    it("should render correct data when incoming data is empty string", () => {

        let component = shallow(<DataView title="Capital" data="" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('n/a');
    });

    it("should render correct data when incoming data is 'Kiev'", () => {

        let component = shallow(<DataView title="Capital" data="Kiev" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('Kyiv');
    });

    it("should render correct data with custom incoming data", () => {

        let component = shallow(<DataView title="Capital" data="Paris" position="" addictions=""/>);
        expect(component.find(".info__content").text()).toEqual('Paris');
    });

});

