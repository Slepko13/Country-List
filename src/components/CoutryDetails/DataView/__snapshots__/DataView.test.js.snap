// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my DataView component should render DataView  Component with props(snapshot) 1`] = `
<div
  className="DataView"
>
  <div
    className="info__title"
  >
    Capital
  </div>
  <div
    className="info__content"
  >
    Kyiv
    <span
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    />
  </div>
</div>
`;

exports[`my DataView component should render DataView  Component without props(snapshot) 1`] = `
<div
  className="DataView"
>
  <div
    className="info__title"
  />
  <div
    className="info__content"
  >
    <span
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    />
  </div>
</div>
`;

exports[`my DataView component should render DataView  additional classname depending on props  1`] = `
<div
  className="DataView + left "
>
  <div
    className="info__title"
  />
  <div
    className="info__content"
  >
    <span
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    />
  </div>
</div>
`;
